### DISCUSSION TITLE
MVP for regression task

### DISCUSSION THREAD

**Discussion by Roxyi** [timeseriesAI/tsai#368] [Q&A] [answered]
Hi, the default mask of `MVP` is `subsequence_mask`, which may result in data leakage in a regression task(?). If the downstream task is a regression task, I should set `future_mask` to `True` and `subsequence_mask ` to `False` to pretrain an MVP. Am I right?

**Comment by oguiza** [comment, marked as answer]
Great question, @Roxyi!
The only way to know it is to try it and see the performance in the test set.
However, I'd say that if your task is truly a regression task I don't think there's a risk of leakage and you could use the default mask. If your task is a forecasting task, then there's a risk, and you should use a future mask.
By a regression task, I mean one where the label is a continuous value for each univariate or multivariate time series. For example, if you need to estimate the user's age based on his/her resting heart rate. In this case, the label is not contained in the time series.

**Reply by Roxyi** [reply]
Thank you! I will try it.
